test/integration/apiserver/apply/apply_crd_test.go

```diff
@@ -720,3 +723,113 @@ spec:
+func TestApplyCRDStatus(t *testing.T) {
+	defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.ServerSideApply, true)()
+
+	server, err := apiservertesting.StartTestServer(t, apiservertesting.NewDefaultTestServerOptions(), nil, framework.SharedEtcd())
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer server.TearDownFn()
+	config := server.ClientConfig
+
+	apiExtensionClient, err := clientset.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	dynamicClient, err := dynamic.NewForConfig(config)
+	if err != nil {
+		t.Fatal(err)
+	}
+	noxuDefinition := fixtures.NewMultipleVersionNoxuCRD(apiextensionsv1beta1.ClusterScoped)
+
+	var c apiextensionsv1beta1.CustomResourceValidation
+	err = json.Unmarshal([]byte(`{
+		"openAPIV3Schema": {
+			"type": "object",
+			"properties": {
+				"spec": {
+					"type": "object",
+					"properties": {
+						"ports": {
+							"type": "integer"
+						}
+					}
+				},
+				"status": {
+					"type": "object",
+					"properties": {
+						"conditions": {
+							"type": "array",
+							"x-kubernetes-list-type": "map",
+							"items": {
+								"type": "integer"
+							}
+						}
+					}
+				}
+			}
+		}
+	}`), &c)
+	if err != nil {
+		t.Fatal(err)
+	}
+	noxuDefinition.Spec.Validation = &c
+	falseBool := false
+	noxuDefinition.Spec.PreserveUnknownFields = &falseBool
+	noxuDefinition.Spec.Subresources = &apiextensionsv1beta1.CustomResourceSubresources{
+		Status: &apiextensionsv1beta1.CustomResourceSubresourceStatus{},
+	}
+
+	noxuDefinition, err = fixtures.CreateNewCustomResourceDefinition(noxuDefinition, apiExtensionClient, dynamicClient)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	kind := noxuDefinition.Spec.Names.Kind
+	apiVersion := noxuDefinition.Spec.Group + "/" + noxuDefinition.Spec.Version
+	name := "mytest"
+
+	client := dynamicClient.Resource(schema.GroupVersionResource{
+		Group:    noxuDefinition.Spec.Group,
+		Version:  noxuDefinition.Spec.Version,
+		Resource: noxuDefinition.Spec.Names.Plural,
+	})
+
+	yamlBody := []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+spec:
+  ports: 0
+`, apiVersion, kind, name))
+	obj, err := client.Patch(context.TODO(), name, types.ApplyPatchType, yamlBody, metav1.PatchOptions{FieldManager: "apply_test"})
+	if err != nil {
+		t.Fatalf("failed to create custom resource with apply: %v:\n%v", err, obj)
+	}
+	yamlBody = []byte(fmt.Sprintf(`
+apiVersion: %s
+kind: %s
+metadata:
+  name: %s
+status:
+  conditions:
+  - 0
+`, apiVersion, kind, name))
+	obj, err = client.Patch(context.TODO(), name, types.ApplyPatchType, yamlBody, metav1.PatchOptions{FieldManager: "apply_status_test"})
+	if err != nil {
+		t.Fatalf("failed to apply status: %v:\n%v", err, obj)
+	}
+	accessor, err := meta.Accessor(obj)
+	if err != nil {
+		t.Fatalf("Failed to get meta accessor: %v:\n%v", err, obj)
+	}
+
+	if managedFields := accessor.GetManagedFields(); len(managedFields) != 2 {
```

Comment:
I was expecting this to fail, according to #88901, but this is correct (and I checked the actual object, it has both managers with the right fields).